File: triptracking-library/flutter_plugin/ios/Classes/triptrackingPlugin.swift
import Flutter
import triptracking

public class triptrackingPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "com.carmd.triptracking/flutter",
            binaryMessenger: registrar.messenger()
        )
        let instance = triptrackingPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "doSomething":
            let args = call.arguments as! [String: Any]
            let input = args["input"] as? String ?? ""
            result(triptracking().doSomething(input: input))
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: triptracking-library/flutter_plugin/android/src/main/kotlin/com/carmd/triptracking/flutter/triptrackingPlugin.kt
package com.carmd.triptracking.flutter

import com.carmd.triptracking.triptracking
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class triptrackingPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(binding.binaryMessenger, "com.carmd.triptracking/flutter")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "doSomething" -> {
                val input = call.argument<String>("input") ?: ""
                result.success(triptracking().doSomething(input))
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
